pkg/scheduler/apis/config/types.go

```diff
@@ -106,6 +106,13 @@ type KubeSchedulerConfiguration struct {
 	// Extenders are the list of scheduler extenders, each holding the values of how to communicate
 	// with the extender. These extenders are shared by all scheduler profiles.
 	Extenders []Extender
+
+	// PreemptorOnNominatedNode tells scheduler whether it is allowed to short-circuit to check the
+	// nominated node directly instead of loop over all the nodes in the cluster.
+	// Set this to true implies the preemptor pod might not be dispatched to the best candidate in
+	// some corner case, e.g. another node releases the resources and hence is the best candidate
+	// while the victims got removed from the nominated node.
+	PreemptorOnNominatedNode bool
```

Comment:
I still think this shouldn't be an option. But if people strongly disagree, let's put it in the preemption plugin args.

Reply:
Oops, sorry. I guess this is not related to the preemption plugin. I would call it `TryNominatedNodeFirst`

Reply:
How about the non-optimal case as what @Huang-Wei 's concern if make it as default?

Reply:
partially done,  has been renamed to `TryNominatedNodeFirst`

Reply:
I think the non-optimal case would be rare.

Reply:
I agree it's rare, maybe we can make it as an option first for a while and deprecate the option after the couple of cycle if there is no negative feedback from end-user?

Reply:
Done